sonar-text-plugin/src/test/java/org/sonar/plugins/secrets/utils/UpdatingSpecificationFilesGenerator.java

```diff
@@ -0,0 +1,253 @@
+@SuppressWarnings("java:S3577")
+class UpdatingSpecificationFilesGenerator {
+
+  private final Charset charset = StandardCharsets.UTF_8;
+  private static final Logger LOG = Loggers.get(UpdatingSpecificationFilesGenerator.class);
+
+  @Test
+  @SuppressWarnings("java:S2699")
+  void firstStep() {
+    writeSpecificationFileDefinition();
+  }
+
+  @Test
+  void secondStep() {
+
+    SpecificationLoader specificationLoader = new SpecificationLoader();
+
+    List<String> listOfAlreadyExistingKeys = retrieveAlreadyExistingKeys();
+    Set<String> setOfExistingCheckClassNames = SecretsRulesDefinition.checks().stream().map(Class::getSimpleName).collect(Collectors.toSet());
+
+    Map<String, List<Rule>> rulesMappedToKey = specificationLoader.getRulesMappedToKey();
+
+    Set<String> keysToImplementChecksFor = rulesMappedToKey.keySet();
+    keysToImplementChecksFor.removeAll(listOfAlreadyExistingKeys);
+
+    List<String> newCheckNames = new ArrayList<>();
+    for (String rspecKey : keysToImplementChecksFor) {
+      String checkName = rulesMappedToKey.get(rspecKey).get(0).getProvider().getMetadata().getName();
+      checkName = sanitizeCheckName(checkName, setOfExistingCheckClassNames);
+      writeCheckFile(checkName, rspecKey);
+      writeCheckTestFile(checkName);
+      newCheckNames.add(checkName);
+    }
+
+    writeUpdatedRulesDefinition(newCheckNames);
+  }
+
+  List<String> retrieveAlreadyExistingKeys() {
+    List<String> rspecKeys = new ArrayList<>();
+
+    List<Class<?>> checks = SecretsRulesDefinition.checks();
+    for (Class<?> check : checks) {
+      Constructor<?>[] declaredConstructors = check.getDeclaredConstructors();
+      Check instantiatedCheck;
+      try {
+        instantiatedCheck = (Check) declaredConstructors[0].newInstance();
+      } catch (Exception e) {
+        LOG.error("Error while retrieving already existing keys");
+        throw new RuntimeException(e);
+      }
+      rspecKeys.add(instantiatedCheck.ruleKey.rule());
+    }
+
+    return rspecKeys;
+  }
+
+  private String sanitizeCheckName(String checkName, Set<String> existingClassNames) {
+    checkName = checkName.replaceAll("[^a-zA-Z]", "");
+    checkName = checkName + "Check";
+
+    if (existingClassNames.contains(checkName)) {
+      checkName = checkName.replace("Check", "UniqueNameCheck");
+    }
+    existingClassNames.add(checkName);
+    return checkName;
+  }
+
+  private void writeCheckFile(String checkName, String rspecKey) {
+    Path checkPath = Path.of("src", "main", "java", "org", "sonar", "plugins", "secrets", "checks", checkName + ".java");
+    Path checkTemplatePath = Path.of("src", "test", "resources", "templates", "GenericCheckTemplate.java");
+    try {
+      Files.copy(checkTemplatePath, checkPath);
+
+      String content = Files.readString(checkPath, charset);
+      content = content.replaceAll("GenericCheckTemplate", checkName);
+      content = content.replaceAll("<RSPEC-KEY>", rspecKey);
```

Comment:
Why `replaceAll` instead of `replace`? The first argument is not a regex, so `replace` will be faster.